File: Models/DAL/conexion/clsMyConnection.vb
'Imports System.Data.SqlClient

'' CLASE: clsMyConnection    @PedroSerrano   License: OpenSource
''
'' Esta clase contiene los métodos necesarios para trabajar con el acceso a una base de datos SQL desde una aplicación de Visual Basic
''
'' PROPIEDADES
''   _database: cadena, básica. Consultable/modificable.
''   _user: cadena, básica. Consultable/modificable.
''   _pass: cadena, básica. Consultable/modificable.
''
'' MÉTODOS
''   Function getConnection() As SqlConnection
''       Este método abre una conexión con la base de datos. Lanza excepciones de tipo: SqlExcepion, InvalidOperationException y Exception.
''   
''   Sub closeConnection(ByRef connection As SqlConnection)
''       Este método cierra una conexión con la base de datos. Lanza excepciones de tipo: SqlExcepion, InvalidOperationException y Exception.
''
''   Sub closeDataReader(ByRef myDataReader As SqlDataReader)
''       Este método cierra un objeto de la clase SqlDataReader. Lanza excepciones de tipo: SqlExcepion, InvalidOperationException y Exception.
''

'Public Class clsMyConnection

'#Region "Atributos"
'    Private _database As String
'    Private _user As String
'    Private _pass As String
'#End Region

'#Region "Constructores"

'    'Constructor por defecto
'    Public Sub New()
'        Me._database = "ProductsDB"
'        Me._user = "prueba"
'        Me._pass = "123"
'    End Sub

'    'Constructor con parámetros
'    Public Sub New(ByVal database As String, ByVal user As String, ByVal pass As String)
'        Me._database = database
'        Me._user = user
'        Me._pass = pass
'    End Sub

'#End Region

'#Region "Consultores y modificadores"

'    Public Property Database() As String
'        Get
'            Return _database
'        End Get
'        Set(ByVal value As String)
'            _database = value
'        End Set
'    End Property

'    Public Property User() As String
'        Get
'            Return _user
'        End Get
'        Set(ByVal value As String)
'            _user = value
'        End Set
'    End Property

'    Public Property Pass() As String
'        Get
'            Return _pass
'        End Get
'        Set(ByVal value As String)
'            _pass = value
'        End Set
'    End Property


'#End Region

'#Region "Métodos"

'    ''' <summary>
'    ''' This method open a connection with the database.
'    ''' </summary>
'    ''' <pre>Nothing.</pre>
'    ''' <post>The connection is opened.</post>
'    ''' <returns>A connection with the database.</returns>
'    ''' <remarks></remarks>
'    Public Function getConnection() As SqlConnection

'        Dim connection As New SqlConnection()

'        Try
'            'connection.ConnectionString = "Data Source=" & My.Computer.Name & "Initial Catalog=" & _database & ";uid=" & _user & ";pwd=" & _user & ";"
'            connection.ConnectionString = "server=(local);database=" & _database & ";uid=" & _user & ";pwd=" & _pass & ";"

'            connection.Open()
'        Catch exSql As SqlException
'            Throw exSql
'        Catch exInOp As InvalidOperationException
'            Throw exInOp
'        Catch ex As Exception
'            Throw ex
'        End Try

'        Return connection

'    End Function

'    ''' <summary>
'    ''' This method close a connection with the database.
'    ''' </summary>
'    ''' <pre>Nothing.</pre>
'    ''' <post>The connection is closed.</post>
'    ''' <param name="connection">The input parameter is the connection to close.</param>
'    ''' <remarks></remarks>
'    Public Sub closeConnection(ByRef connection As SqlConnection)

'        Try
'            connection.Close()
'        Catch exSql As SqlException
'            Throw exSql
'        Catch exInOp As InvalidOperationException
'            Throw exInOp
'        Catch ex As Exception
'            Throw ex
'        End Try
'    End Sub


'#End Region

'End Class

File: Models/DAL/conexion/clsMyConnection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Data.SqlClient;

namespace ResidentEvil.Models.DAL.conexion
{
    public class clsMyConnection
    {
        #region atributos
            public String database { get; set; }
            public String user { get; set; }
            public String pass { get; set; }
        #endregion

        #region constructores
            //constructor por defecto
            public clsMyConnection()
            {
                this.database = "Resident Evil Casa";
                this.user = "prueba";
            //    this.pass = "prueba";
                this.pass = "prueba123";
            }
            //constructor con parámetros
            public clsMyConnection(String database, String user, String pass)
            {
                this.database = database;
                this.user = user;
                this.pass = pass;
            }
        #endregion
            /**
             * ''' This method open a connection with the database.
//'    ''' </summary>
//'    ''' <pre>Nothing.</pre>
//'    ''' <post>The connection is opened.</post>
//'    ''' <returns>A connection with the database.</returns>
//'    ''' <remarks></remarks>
             */
            public SqlConnection getConnection()
            {
                SqlConnection connection = new SqlConnection();
                try
                {
                    
                    //connection.ConnectionString = "Data Source=" + System.Environment.MachineName + "Initial Catalog=" + database + ";uid=" + user + ";pwd=" + user + ";";
            //        connection.ConnectionString = "server=(local);database=" + database + ";uid=" + user + ";pwd=" + pass + ";";
            //        connection.ConnectionString = "server=PRIVADO-PC\\MIEQUIPO;database=" + database + ";uid=" + user + ";pwd=" + pass + ";";
                    connection.ConnectionString = "server=PRIVADO-PC\\SEKUL;database=" + database + ";uid=" + user + ";pwd=" + pass + ";";
                    connection.Open();
                }
                catch(SqlException){
                    throw;
                }
                return connection;
            }

            /*
             * ''' <summary>
    '    ''' This method close a connection with the database.
    '    ''' </summary>
    '    ''' <pre>Nothing.</pre>
    '    ''' <post>The connection is closed.</post>
    '    ''' <param name="connection">The input parameter is the connection to close.</param>
    '    ''' <remarks></remarks>
             */
            public void closeConnection(ref SqlConnection connection)
            {
                try
                {
                    connection.Close();
                }
                catch (SqlException)
                {
                    throw;
                }
                catch (Exception)
                {
                    throw;
                }
            }

        
    }
}